### PR TITLE
Fix a BUNCH of little glitches in my bad bot auth changes

### PR DESCRIPTION
**Pull request by statik**
Pretty sure I proposed busted code to merge, due to some subtle differences in the way the app auth library code works that I missed on the first pass.

This code is much more likely to work, there is a js version runnning successfully at https://github.com/kindlyops/examples/runs/1629296613?check_suite_focus=true.

I also attempted some improvements like using the core functions for masking, success, failure.

### CODE REVIEW

File: cmd/bot/index.ts [hunk 1]
```diff
@@ -13,32 +14,34 @@ async function main() {
 
   if (env.CHECK === "yes") {
     console.log("ok");
-    return;
+    return "ok";
```
Comment (whilp) [inline cmd/bot/index.ts:17]: I think we should update main()'s signature as well (async main() string)

**Review by whilp — APPROVED** [review]
Thanks! I've pored over this flow in go, js, and python and still have a hard time mapping those SDKs to the REST API. It doesn't help that find installation by repository is not documented....

**Comment by whilp** [comment]
Btw, see also WIP in https://github.com/whilp/world/pull/308 to extend cmd/gh to handle auth.

**Comment by statik** [comment]
> Btw, see also WIP in #308 to extend cmd/gh to handle auth.

very cool! I'm happy about not having to track installation ids around between every repo now

File: cmd/bot/index.ts (hunk 1 above)
Comment (whilp) [reply cmd/bot/index.ts:17]: Oh yeah is promise

**Review by whilp — APPROVED** [review]

**Comment by whilp** [comment]
@statik  fwiw I noticed that my scheduled renovate runs (update action) started to fail. Eg

https://github.com/whilp/world/actions/runs/454992101

I’m debugging and poking at a fix over in https://github.com/whilp/world/pull/645

**Pull request merged**